power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -528,23 +528,27 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         // cache component update order if possible
         bool const is_independent = MainModelImpl::is_update_independent(update_data);
 
-        return [&base_model, &exceptions, &infos, &calculation_fn, &result_data, &update_data,
+        std::shared_ptr<SequenceIdx const> all_scenarios_sequence{
+            is_independent ? std::make_shared<SequenceIdx const>(get_sequence_idx_map(update_data)) : nullptr};
+
+        return [&base_model, &exceptions, &infos, &calculation_fn, &result_data, &update_data, all_scenarios_sequence,
                 is_independent](Idx start, Idx stride, Idx n_scenarios) {
             assert(n_scenarios <= narrow_cast<Idx>(exceptions.size()));
             assert(n_scenarios <= narrow_cast<Idx>(infos.size()));
+            assert(is_independent == (all_scenarios_sequence != nullptr));
 
             Timer const t_total(infos[start], 0000, "Total in thread");
 
-            auto copy_model = [&base_model, &infos](Idx scenario_idx) {
+            auto const copy_model = [&base_model, &infos](Idx scenario_idx) {
```

Comment:
As we agreed. Remove this named lambda. Just create the lambda and immediately execute it. The name `copy_model` is confusing.

Reply:
then name it like `copy_model_functor`